### PR TITLE
slack with google auth

### PR DESCRIPTION
**Pull request by dubchuck**
## Ticket
https://app.asana.com/0/1201247707079024/1202809255712150

## Goal
To make a reusable workflow that can post to slack using a slack bot api token retrieved from Google Secrets.

## Code Changes
- Added test workflow
- Added actual shared workflow

## Related Pull Requests
https://github.com/HausAnalytics/regiontest/pull/34

## Notes & Questions for Reviewers
- Is this the right approach for passing secrets, inputs into a shared repository?

### Checks
- [X] Changes in this PR follows Haus' [code contribution guidelines](https://docs.google.com/document/d/1-3pOkN0Qk5OrsoDJCFPL_Jwunc6ptXLzSpTdTRAH7to/edit#heading=h.fy8j7l7xqyx8) or if it breaks guidelines I've explained why it's necessary.
- [ ] I'm making a vmajor or vminor [semver](https://semver.org/) change and have that present in the branch name

### CODE REVIEW

File: .github/workflows/post_to_slack_google_secret.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    timeout-minutes: 1
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+    # Authorize with passed in GOOGLE CREDENTIAL JSON
+    - name: Auth with GCP
+      uses: 'google-github-actions/auth@v0'
+      with:
+        credentials_json: '${{ secrets.GOOGLE_CREDENTIALS_FOR_SLACK_TOKEN }}'
```
Comment (drorasaf) [inline .github/workflows/post_to_slack_google_secret.yml:30]: according to the documentation, they recommend using the workload identity, is it possible to use it here instead(in some cases the workload identity pool is still not working)?

File: .github/workflows/test_post_to_slack_google_secret.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: Test post to slack with token from google secrets
+
+on:
+  push
+
+
+jobs:
+  test_post_to_slack:
+    uses: HausAnalytics/actions/.github/workflows/post_to_slack_google_secret.yml@f-slack
```
Comment (drorasaf) [inline .github/workflows/test_post_to_slack_google_secret.yml:9]: can use:
```
 ./.github/workflows/post_to_slack_google_secret.yml
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+TEXT ❗
+Text you want to send to the channel
+
+BLOCKS
+Encoded JSON string similar to what is outlined here https://api.slack.com/reference/block-kit/blocks
+
+Secrets
+
+These secrets must be exist in the calling repo
```
Comment (drorasaf) [inline README.md:19]: must be exist -> must exist

File: README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Usage
+```
+name: Test post to slack with token from google secrets
+
+on:
+  push
+
+jobs:
+  test_post_to_slack:
+    uses: HausAnalytics/actions/.github/workflows/post_to_slack_google_secret.yml@f-slack
```
Comment (drorasaf) [inline README.md:36]: we should remove the branch name

File: README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    secrets: inherit
+    with:
+      CHANNEL_ID: "#test_channel"
+      TEXT: "Test Text"
```
Comment (drorasaf) [inline README.md:40]: Care to add an example with the block?

File: .github/workflows/post_to_slack_google_secret.yml (hunk 1 above)
Comment (dubchuck) [reply .github/workflows/post_to_slack_google_secret.yml:30]: Can we add that as an enhancement for the next version?  I'd like to finalize migration of regiontest integration and update them after they are both merged.

File: .github/workflows/post_to_slack_google_secret.yml [hunk 6]
```diff
@@ -27,14 +34,16 @@ jobs:
     - name: Auth with GCP
       uses: 'google-github-actions/auth@v0'
       with:
-        credentials_json: '${{ secrets.GOOGLE_CREDENTIALS_FOR_SLACK_TOKEN }}'
+        create_credentials_file: true
```
Comment (drorasaf) [inline .github/workflows/post_to_slack_google_secret.yml:37]: I'm not sure this is needed, I got some warnings when this was one saying there is no need for it if we don't reuse it later

File: .github/workflows/post_to_slack_google_secret.yml [hunk 7]
```diff
@@ -11,11 +11,18 @@ on:
...
         description: Simple text message sent with Slack message
-      BLOCKS:
+      WORKLOAD_IDENTITY_PROVIDER:
+        required: true
+        type: string
+        description: Identity Provider
+      SERVICE_ACCOUNT:
+        required: false
+        type: string
+        description: Service account
```
Comment (drorasaf) [inline .github/workflows/post_to_slack_google_secret.yml:21]: nit: maybe add that the service account must have secret manager viewer access to the secret for the slack bot

**Review by drorasaf — APPROVED** [review]

File: .github/workflows/post_to_slack_google_secret.yml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    - name: Checkout code
+      uses: actions/checkout@v3
+    # Authorize with passed in GOOGLE CREDENTIAL JSON
+    - name: Auth with GCP
+      uses: 'google-github-actions/auth@v0'
+      with:
+        create_credentials_file: true
+        workload_identity_provider: '${{ inputs.WORKLOAD_IDENTITY_PROVIDER }}'
+        service_account: '${{ inputs.SERVICE_ACCOUNT }}'
+    # Retrieve secret from Google Secrets using
```
Comment (leebs778) [inline .github/workflows/post_to_slack_google_secret.yml:40]: Using…?

File: .github/workflows/post_to_slack_google_secret.yml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    - name: Fetch secret from Google Secrets to deploy message using slack
+      id: google_secret
+      uses: 'google-github-actions/get-secretmanager-secrets@v0'
+      with:
+        secrets: |-
+            slack_token:${{ inputs.GOOGLE_SECRET_KEY_FOR_SLACK_TOKEN }}
+    # Use slack token that is retrieved from google secrets to post to slack
+    - name: Post to slack channel with only text
+      if: "github.event.inputs.BLOCKS == ''"
```
Comment (leebs778) [inline .github/workflows/post_to_slack_google_secret.yml:49]: is this a valid check? I don’t see blocks defined as an input

File: README.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
+Post to Slack using Google Secret
+
+Create a label in your repo to assign to stuck pull requests. The default label this action uses is stuck, but you can use any label.
```
Comment (leebs778) [inline README.md:3]: Copy pasta?

**Review by leebs778 — COMMENTED** [review]
Is the fetching of the relevant pr info staying out of scope here?

**Pull request merged**